spec/Appendix A -- Field Selection.md

```diff
@@ -0,0 +1,549 @@
+
+```json
+{
+    "data": {
+        "userById": {
+            "id": "123"
+        }
+    }
+}
+```
+
+The {FieldSelectionMap} scalar type is used to establish semantic equivalence between an argument and
+the fields within the associated return type. To accomplish this, the scalar must define the
+relationship between input fields or arguments and the fields in the resulting object. 
+Consequently, a {FieldSelectionMap} only makes sense in the context of a specific argument and its 
+return type.
+
+The {FieldSelectionMap} scalar is represented as a string that, when parsed, produces a {SelectedValue}.
+
+A {SelectedValue} must exactly match the shape of the argument value to be considered
+valid. For non-scalar arguments, you must specify each field of the input type in
+{SelectedObjectValue}.
+
+```graphql example
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "{ firstName: firstName }")): User @lookup
+}
+```
+
+```graphql counter-example
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "firstName")): User @lookup
+}
+```
+
+
+## Language
+
+According to the GraphQL specification, an argument is a key-value pair in which the key is the name
+of the argument and the value is a `Value`.
+
+The `Value` of an argument can take various forms: it might be a scalar value (such as `Int`,
+`Float`, `String`, `Boolean`, `Null`, or `Enum`), a list (`ListValue`), an input object
+(`ObjectValue`), or a `Variable`.
+
+Within the scope of the {FieldSelectionMap}, the relationship between input and output is
+established by defining the `Value` of the argument as a selection of fields from the output object.
+
+Yet only certain types of `Value` have a semantic meaning. 
+`ObjectValue` and `ListValue` are used to define the structure of the value. 
+Scalar values, on the other hand, do not carry semantic importance in this context, and variables
+are excluded as they do not exist. 
+Given that these potential values do not align with the standard literals defined in the GraphQL
+specification, a new literal called {SelectedValue} is introduced, along with {SelectedObjectValue},
+
+Beyond these literals, an additional literal called {Path} is necessary.
+
+### Name
+
+Is equivalent to the {Name} defined in the 
+[GraphQL specification](https://spec.graphql.org/October2021/#Name)
+
+### Path
+Path :: 
+    - FieldName
+    - Path . FieldName
+    - FieldName < TypeName > . Path
+
+FieldName ::
+    - Name
+
+TypeName ::
+    - Name
+
+The {Path} literal is a string used to select a single output value from the _return type_ by
+specifying a path to that value. 
+This path is defined as a sequence of field names, each separated by a period (`.`) to create 
+segments. 
+
+``` example
+book.title
+```
+
+Each segment specifies a field in the context of the parent, with the root segment referencing a
+field in the _return type_ of the query. 
+Arguments are not allowed in a {Path}.
```

Comment:
Arguments are currently supported for `@requires` in Federation: https://www.apollographql.com/docs/federation/entities-advanced/#using-requires-with-fields-that-take-arguments

For `@is` I agree that arguments do not need to be supported, if we are reusing this Appendix for other directives we could change this to a `MAY be used but are context dependent`